socranop/installtool.py

```diff
@@ -687,47 +697,46 @@ def shutdown_service(self, reason):
         if not dbus_service.NameHasOwner(const.BUSNAME):
             common.debug("D-Bus service not running")
         else:
-            self._print_heading(reason)
             service = self._service(const.BUSNAME)
             service_version = service.version
-            with Step("stop", f"Shutting down version {service_version}"):
+            with Step(
+                "stop",
+                f"Shutting down running service version {service_version}",
+                max_attempts=MAX_ATTEMPTS,
+            ) as step:
                 service.Shutdown()
+                # Wait until the shutdown clears the service off the bus
+                while dbus_service.NameHasOwner(const.BUSNAME):
+                    step.try_again()
 
     def verify_install(self, force=False):
         if self.no_launch and not force:
             common.debug("no_launch is set; not verifying our dbus service")
             return
 
-        self._print_heading("Verifying")
-
         dbus_service = self._service(".DBus")
         with Step("verify", "Reload dbus to register the new service"):
             # An explicit reload is required in case dbus isn't smart enough to see
             # us drop in the service file (dbus-broker, for example, if we just
             # created ~/.local/share/dbus-1/services)
             dbus_service.ReloadConfig()
 
-        max_attempt = 5
         with Step(
             "verify",
             "Checking for registered service",
-            error_msg=f"The dbus service we just installed has not yet been detected after {max_attempt}s. You may need to restart your dbus session (for example, by logging out and back in to your desktop).",
+            error_msg=f"The dbus service we just installed has not yet been detected after {MAX_ATTEMPTS}s. You may need to restart your dbus session (for example, by logging out and back in to your desktop).",
```

Comment:
And here it would maybe be better to use a local, or make the error message generic, since the error will have the attempt count appended anyway.

Reply:
Before this, we had a "5s" somewhere in a string, and a `max_attempts=5` somewhere in code. Now we have a single place where the 5 can be changed, and both the string and the value code change in sync. If you want to change the error message, that is a different change.